# Statistical Significance
#### Hypothesis Testing
- Hypothesis testing = A technique for evaluating a theory using data
    + The hypothesis = initial belief about the situation before the study
        + Null hypothesis = common accepted facts
        + Alternate hypothesis = the opposite of Null hypothesis
    + The researcher’s task is to reject/disprove the null hypothesis

- Example: A study about cell phones and cancer risk
    + **Problem**
        + Null hypothesis: “Cell phones have no effect on cancer risk.”
        + Alternative hypothesis: “Cell phones affect the risk of cancer.”
    - **Solution**
        + Common goal = determine which of the 2 hypotheses is better supported bydata
        + Common hypothesis tests = make comparisons 
            + Between two groups 
            + Or between one group and the entire population

#### Z-test
- Z-test = a statistical technique to test the Null Hypothesis against the Alternate Hypothesis. Used when
    + sample data is normally distributed 
    + Population size is greater than 30
        + According to the Central Limit Theorem, the samples are considered to be normally distributed whenever sample size exceeds 30

- $Z = \frac{\bar{x}-\mu}{\sigma/ \sqrt{n}}$
    + $\bar{x}$: mean of sample
    + $\mu$: mean of population
    + $\sigma$: std of population
    + $n$: number of observations

#### P-value
- p-value quantifies the rareness in our results
    + Tells us how often we’d see the results of an experiment 
    + We can use p-values to reach conclusions in significance testing

<img src="./img/11.png" width="500"/>

+ We compare the p-value to a significance level ($\alpha$) to make conclusions about our hypotheses
    + If $\text{p-value} < \alpha$ = the result would rarely occur by chance
        + The result is statistically significant
        + We can reject the null hypothesis in favor of the alternative hypothesis
    + If $\text{p-value} \geq \alpha$
        + Fail to reject the null hypothesis
        + This doesn’t mean we accept the null hypothesis though!

+ How to choose $\alpha$?
    + Choice of $\alpha$ depends on the situation
    + 0.05 = the most widely used value across all scientific disciplines
    - $p<0.05$ meaning
        + there is less than 5% chance of seeing our results, in the world where the null hypothesis is true
    - $p<0.05$ **does not** mean
        + there’s less than 5% chance that our experimental results are due to random chance

<p align="center">
    <img src="./img/12.png" width="220"/>
    <br/>
    <i><a href=https://xkcd.com//>Credit</a></i>
</p>

## Significance Testing Example 1
- **Problem**: A company claims that it has a high hiring bar which its employees having an IQ above the average
    + A random sample of their 40 employees has a mean IQ score of 115
    + Given mean population IQ is 100 with a standard deviation of 15?
    + Is this sufficient evidence to support the company’s claim?
- **Solution**
    + State the Null hypothesis
        + H0 - 40 employees have population mean IQ = 100: $\mu = 100$
    + State the Alternate Hypothesis:
        + H1 - 40 employees have above population average IQ scores: $\mu > 100$
    + State the threshold for the p-value
        + $\alpha = 0.05$
    + Calc z-test
        + $Z = \frac{\bar{x}-\mu}{\sigma/ \sqrt{n}} = \frac{115-110}{15/\sqrt{40}} = 6.32$
    + Calc p-value from z-score

In [1]:
import scipy.stats as ss

z_scores = 6.32
p_values = ss.norm.sf(abs(z_scores))
p_values

1.3078165132642286e-10

- **p = extremely small** meaning
    + If 40 employees have population mean IQ (H0 is true), there is an extremely small **random** chance that a sample of 40 employees will have IQ mean = 115
    
- Conclude
    + p < 0.05 => we can reject the null hypothesis
    + 40 employees have an unusually higher IQ score compared to random samples of similar size from the entire population

## Significance Testing Example 2
- **Problem**: A study that tests the impact of smoking on the duration of pregnancy. 
    + We randomly sample 40 women who smoked, has mean duration of pregnancy = 260 days
    + While the mean pregnancy length = 266 days with std = 21 days
    + Is this sufficient evidence to support the company’s claim?
- **Solution**
    + State the Null hypothesis
        + H0 - smoking has no effect on the duration of pregnancy, 40 smoking women have population mean = 266 days: $\mu = 266$
    + State the Alternate Hypothesis:
        + H1 - smoking impact on the duration of pregnancy: $\mu < 266$
    + State the threshold for the p-value
        + $\alpha = 0.05$
    + Calc z-test
        + $Z = \frac{\bar{x}-\mu}{\sigma/ \sqrt{n}} = \frac{260-266}{21/\sqrt{40}} = -1.81$
    + Calc p-value from z-score

In [2]:
import scipy.stats as ss

z_scores = -1.81
p_values = ss.norm.sf(abs(z_scores))
p_values

0.03514789358403879

- **p = 0.035** meaning
    + If smoking has no effect on the duration of pregnancy (H0 is true), there is a 3.5% **random** chance that a sample of 40 smokers will have pregnancies lasting less than 260 days

- Conclude
    + p < 0.05 => we can reject the null hypothesis
    + 40 smoking women have an unusually duration of pregnancy compared to random samples of similar size from the entire population